### PR TITLE
Docs: expand deploy recipe

### PR DESCRIPTION
**Pull request by tom-raley**
## Description
Expands the Deploy section in recipes and updates it to the new format

## Related Issues
Addresses [#15319](https://github.com/gatsbyjs/gatsby/issues/15319)

### CODE REVIEW

File: docs/docs/recipes.md [hunk 1]
```diff
@@ -122,12 +122,54 @@ To wrap pages with layouts, use normal React components.
 
 ## Deploying
 
-Showtime.
+Showtime. Once you are happy with your site, you are ready to go live with it!
+
+#### Prerequisites
+
+-A Gatsby site ([check out the Tutorial if you don't yet have a Gatsby site](/tutorial/))
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
-A [Gatsby site](/docs/quick-start)
```
i've been linking to the Quick Start to keep recipes focused.

File: docs/docs/recipes.md [hunk 2]
```diff
@@ -122,12 +122,54 @@ To wrap pages with layouts, use normal React components.
...
+#### Prerequisites
+
+-A Gatsby site ([check out the Tutorial if you don't yet have a Gatsby site](/tutorial/))  
+-The [Gatsby CLI](/docs/gatsby-cli)
+
+#### Directions
+
+1. Stop your development server if it is running (`Ctrl + C` on your command line in most cases)
+
+2. If you want the standard path prefix for your site (`/`), run `gatsby build` using the Gatsby CLI on the command line. The files you need to copy to your server will now be in the `public` folder.
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
2. For the standard site path at the root directory (`/`), run `gatsby build` using the Gatsby CLI on the command line. The built files will now be in the `public` folder.
```

File: docs/docs/recipes.md [hunk 3]
```diff
@@ -122,12 +122,54 @@ To wrap pages with layouts, use normal React components.
...
+
+```shell
+gatsby build
+```
+
+3. If you'd like to have the index of your site be at a path other than `/`, set a path prefix by adding the following to your `gatsby-config.js` and replacing `yourpathprefix` with your desired path prefix:
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
3. To include a site path other than `/` (such as `/site-name/`), set a path prefix by adding the following to your `gatsby-config.js` and replacing `yourpathprefix` with your desired path prefix:
```

File: docs/docs/recipes.md [hunk 4]
```diff
@@ -122,12 +122,54 @@ To wrap pages with layouts, use normal React components.
...
+
+```js:title=gatsby-config.js
+module.exports = {
+  pathPrefix: `/yourpathprefix`,
+}
+```
+
+There are a few reasons you might want to do this--for instance, if you are hosting a blog built with Gatsby on a domain where you are running another site not built on Gatsby. The main site would direct to `example.com`, and by setting a path prefix you can have `example.com/blog` load your Gatsby site directly.
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
There are a few reasons to do this--for instance, hosting a blog built with Gatsby on a domain with another site not built on Gatsby. The main site would direct to `example.com`, and the Gatsby site with a path prefix could live at `example.com/blog`.
```
Making this a tad bit shorter for the recipe format!

File: docs/docs/recipes.md [hunk 5]
```diff
@@ -122,12 +122,54 @@ To wrap pages with layouts, use normal React components.
...
+
+Once your path prefix is set in your `gatsby-config.js`, run `gatsby build`, but make sure to add the `--prefix-paths` flag, like so:
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
4. With a path prefix set in `gatsby-config.js`, run `gatsby build` with the `--prefix-paths` flag to automatically add the prefix to the beginning of all Gatsby site URLs and `<Link>` tags.
```

File: docs/docs/recipes.md [hunk 6]
```diff
@@ -122,12 +122,54 @@ To wrap pages with layouts, use normal React components.
...
+
+```shell
+gatsby build --prefix-paths
+```
+
+With this flag added, Gatsby will automatically add your path prefix to the beginning of all of your site's URLs and will automatically update all `<Link>` tags as well.
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
```
Moving to consolidate info in the step above.

File: docs/docs/recipes.md [hunk 7]
```diff
@@ -122,12 +122,54 @@ To wrap pages with layouts, use normal React components.
...
+
+4. It is generally helpful to make sure that your site looks the same when running `gatsby build` as it did when running `gatsby develop`. By running `gatsby serve` after you build your site, you can test out (and debug if necessary) the finished product before deploying it live.
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
5. Make sure that your site looks the same when running `gatsby build` as with `gatsby develop`. By running `gatsby serve` when you build your site, you can test out (and debug if necessary) the finished product before deploying it live.
```

File: docs/docs/recipes.md [hunk 8]
```diff
@@ -122,12 +122,54 @@ To wrap pages with layouts, use normal React components.
...
+
+```shell
+gatsby serve
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
gatsby build && gatsby serve
```

File: docs/docs/recipes.md [hunk 9]
```diff
@@ -122,12 +122,54 @@ To wrap pages with layouts, use normal React components.
...
+```
+
+5. From here, you can either copy the files in your `public` folder directly to your server, or deploy to one of many popular servers. Check out the [deployment docs for specific hosting platforms and how to deploy to them](/docs/deploying-and-hosting/).
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
```
I think it's fine to keep this recipe limited to "preparing for deployment" and let subsequent recipes and Additional resources fill in the details.

File: docs/docs/recipes.md [hunk 10]
```diff
@@ -122,12 +122,54 @@ To wrap pages with layouts, use normal React components.
...
+
+#### Additional Resources
 
 - Walk through building and deploying an example site in [tutorial part one](/tutorial/part-one/#deploying-a-gatsby-site)
-- Learn how to make sure your site is configured properly to be [searchable, shareable, and properly navigable](/docs/preparing-for-site-launch/)
 - Learn about [performance optimization](/docs/performance/)
-- Read about [other deployment related topics](/docs/deploying-and-hosting/)
+- Read about [other deployment related topics](/docs/preparing-for-deployment/)
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
- Check out the [deployment docs](/docs/deploying-and-hosting/) for specific hosting platforms and how to deploy to them
```

**Comment by tom-raley** [comment]
Thanks for the feedback! I tend to err on the side of being more verbose rather than less, which is the whole point of a recipe, so all the feedback is much appreciated!

File: docs/docs/recipes.md [hunk 11]
```diff
@@ -122,13 +122,53 @@ To wrap pages with layouts, use normal React components.
 
 ## Deploying
 
-Showtime.
+Showtime. Once you are happy with your site, you are ready to go live with it!
+
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
### Preparing for deployment
```
I think we're missing a heading for the recipe here!

File: docs/docs/recipes.md [hunk 12]
```diff
@@ -122,13 +122,53 @@ To wrap pages with layouts, use normal React components.
...
+#### Prerequisites
+
+-A [Gatsby site](/docs/quick-start)
+-The [Gatsby CLI](/docs/gatsby-cli)
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
- The [Gatsby CLI](/docs/gatsby-cli) installed
```

File: docs/docs/recipes.md [hunk 13]
```diff
@@ -122,13 +122,53 @@ To wrap pages with layouts, use normal React components.
 
 ## Deploying
 
-Showtime.
+Showtime. Once you are happy with your site, you are ready to go live with it!
+
+#### Prerequisites
+
+-A [Gatsby site](/docs/quick-start)
```
Comment (marcysutton) [inline docs/docs/recipes.md]:
```suggestion
- A [Gatsby site](/docs/quick-start)
```

File: docs/docs/recipes.md (hunk 11 above)
Comment (tom-raley) [reply docs/docs/recipes.md]: Yeah, wasn't sure on the format for that, thanks for adding it in!

**Pull request merged**